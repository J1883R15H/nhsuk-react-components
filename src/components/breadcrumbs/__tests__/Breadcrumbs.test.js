import React from 'react';
import { mount, shallow } from 'enzyme';
import Breadcrumbs from '../breadcrumbs';

describe('Breadcrumbs', () => {
  it('matches snapshot', () => {
    const mountedItem = mount(
      <Breadcrumbs>
        <Breadcrumbs.Item>Test</Breadcrumbs.Item>
      </Breadcrumbs>
    );
    expect(mountedItem).toMatchSnapshot();
    mountedItem.unmount();
  });
  it('item matches snapshot', () => {
    const mountedItem = mount(<Breadcrumbs.Item>Item</Breadcrumbs.Item>);
    expect(mountedItem).toMatchSnapshot();
    mountedItem.unmount();
  });
  it('handles onClick', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Breadcrumbs.Item onClick={onClick}>Item</Breadcrumbs.Item>
    );
    const wrapperForAnonFunc = shallow(
      <Breadcrumbs.Item>Item</Breadcrumbs.Item>
    );
    wrapper.find('a').simulate('click');
    wrapperForAnonFunc.find('a').simulate('click');
    expect(onClick).toHaveBeenCalled();
    wrapper.unmount();
    wrapperForAnonFunc.unmount();
  });
});
