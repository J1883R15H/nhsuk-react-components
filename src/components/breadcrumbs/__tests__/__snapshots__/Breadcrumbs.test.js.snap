// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs item matches snapshot 1`] = `
<Item
  className=""
  style={Object {}}
>
  <li
    className="nhsuk-breadcrumb__item"
    style={Object {}}
  >
    <a
      className="nhsuk-breadcrumb__link"
    >
      Item
    </a>
  </li>
</Item>
`;

exports[`Breadcrumbs matches snapshot 1`] = `
<Breadcrumbs
  className=""
>
  <nav
    className="nhsuk-breadcrumb"
  >
    <div
      className="nhsuk-width-container"
    >
      <ol
        className="nhsuk-breadcrumb__list"
      >
        <Item
          className=""
          style={Object {}}
        >
          <li
            className="nhsuk-breadcrumb__item"
            style={Object {}}
          >
            <a
              className="nhsuk-breadcrumb__link"
            >
              Test
            </a>
          </li>
        </Item>
      </ol>
    </div>
  </nav>
</Breadcrumbs>
`;
